File: sorting/insertionSort.cpp
#include<iostream>
#include<vector>
using namespace std;

// TC : O(n^2)
// SC : O(1)

vector<int> insertionSort(vector<int> arr, int n ){

    for(int i = 1; i < n; i++){
        int curr_element = arr[i];

        int j = i-1;
        while( j >= 0  && arr[j] > curr_element ){
            arr[j+1] = arr[j];
            j--;
        }
        arr[j+1] = curr_element;

    }
    for(int i = 0 ; i < n; i++){
        cout << arr[i] << ' ';
    }
    return arr;
}

int main(){
    int n  = 5;
    vector<int> arr = { 2,5,9,5,4};

    insertionSort(arr , n);



    return 0;
}

File: sorting/InsertionSort.js

function insertionSort(arr){
    //Iteration should start from index 1, bcoz first element we consider Already Sorted !
    let n = arr.length;
    for(let i = 1; i < n ; i++){
        let curr_element = arr[i];// this is tbe element to be Inserted

        // We need to go to the left of the element toill we find the smaller one.

        let j = i-1;
        while( j >=0 && arr[j] > curr_element){
            // till the time you either go beyond the 0th Index or you can find a smaller element
            // we will do Replacements

            arr[j+1] = arr[j];
            j--;
        }
        arr[j+1] =curr_element;

    }
}

let arr = [5,4,3,2,1];
insertionSort(arr);
console.log(arr);